app/workers/scheduled_publishing_job.rb

```diff
@@ -1,10 +1,9 @@
 # frozen_string_literal: true
 
-class ScheduledPublishingWorker
-  include Sidekiq::Worker
-  # We want to retry for up to 5 minutes. 10 retries x 30s intervals = 5 minutes.
-  sidekiq_options retry: 10
-  sidekiq_retry_in { 30 }
+class ScheduledPublishingJob < ApplicationJob
+  retry_on(GdsApi::BaseError,
```

Comment:
This doesn't seem to match up with the number of retries in the previous implementation?

Reply:
I'm happy to keep the original behaviour, but I thought it would be a good to use the more sophisticated retry mechanism provided by ActiveJob. These settings mean we will retry for the same amount of time, but with fewer attempts. The thinking here is similar to protocols like TCP/WiFi, where the idea is to give the remote increasing amounts of time to recover, rather than pummelling it constantly when we know it's already in trouble.

Reply:
Ok, 3 attempts seems a little low to me at first glance but the initial "retry every 30 seconds for up to 5 minutes" was just a first pass at a limit anyway.  I'd find it helpful if there was a comment either in the code or in the commit message to illustrate why we're changing to use `:exponentially_longer` and what that * 3 retry attempts means in terms of time so future devs can more quickly make a decision about how to change that.

Reply:
Turns out I forgot to increase it to 5 anyway, which does give us a 5 minute period. Added a doc too!